From 4499e4566d2e488be1c0e0a20328b13652ed14cb Mon Sep 17 00:00:00 2001
From: kcgen <kcgen@users.noreply.github.com>
Date: Sat, 1 Jan 2022 07:48:27 -0800
Subject: [PATCH] Remove the project's main to prevent masking gmock's main
 (#1476)

clang/llvm ensure gmock's main() function always overrides the main
function coming from the dosbox library, however sometimes with
gcc/ld, gmock's main() doesn't take precedent and instead dosbox's
main is run.

This fixes an issue reported on Fedora and Gentoo where unit tests
were popping up SDL's graphical window, and then either timing our
failing.
---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index d885da1b9..59f677b59 100644
--- a/meson.build
+++ b/meson.build
@@ -444,7 +444,7 @@ executable('dosbox', dosbox_sources,
            include_directories : incdir,
            install : true)
 # create a library so we can test things inside DOSBOX dep path
-libdosbox = static_library('dosbox', ['src/main.cpp', 'src/dosbox.cpp', version_file],
+libdosbox = static_library('dosbox', ['src/dosbox.cpp', version_file],
                         include_directories : incdir,
                         dependencies : [atomic_dep,
                                         threads_dep,
